### ISSUE TITLE
Input looses focus

### ISSUE THREAD

**Issue by pawan-poudel-github** [emilkowalski/vaul#128]
I am using multiple input tags in a drawer, but when I type or click on an input field, it loses focus. However, it works perfectly with a single input tag.

**Comment by emilkowalski** [comment]
Please provide a codesandbox with reproduction

**Comment by pawan-poudel-github** [comment]
> Please provide a codesandbox with reproduction

Here is the sandbox link: 
[https://codesandbox.io/p/sandbox/drawer-with-scale-forked-x5vgtc?file=%2Fapp%2Fmy-drawer.tsx%3A1%2C1-111%2C1](https://codesandbox.io/p/sandbox/drawer-with-scale-forked-x5vgtc?file=%2Fapp%2Fmy-drawer.tsx%3A1%2C1-111%2C1)

**Comment by emilkowalski** [comment]
So what exactly should I be seeing here? Works fine for me, you can record a video as well.

**Comment by michaelmikh** [comment]
@emilkowalski should I open new issue referencing this one? Or tell me how'd you prefer to go on with that

**Comment by Southclaws** [comment]
I'm seeing this too on the latest version, repro'd with pointing device as well as touch on all major OSes. It seems less likely on mobile/touch devices but still happens.

What it feels like is if your touch enter and touch leave positions are different, Vaul is capturing that as a drag event and pulls focus away from the input. It seems especially common (anecdotally) with a mac touchpad or a mouse, presumably because of the slight movement you get from clicking. Could be even more prominent in folks with motor disabilities too as mouse movement between click down/up can be a bit more emphasised.

**Comment by 101budspencer** [comment]
@emilkowalski ?

**Comment by 101budspencer** [comment]
@michaelmikh I think the owner of this repo just doesn't know a fix

**Comment by mrabbas1** [comment]
I have the same issue, I think that's because of vauel-dragging style

**Comment by jsefiani** [comment]
Maybe this helps as a potential workaround:
```
        <Drawer.Content
          onOpenAutoFocus={(e) => {
            e.preventDefault();
          }}
        >
        ...
```

**Comment by mrabbas1** [comment]
Unfortunately, the issue is still there! it's not focused on input

**Comment by mrabbas1** [comment]
> I can’t reproduce in the codesandbox demo above. What browser and device are you on?

OS:macOS 
Device: Mac Mini M1 
browsers: Safari, Firefox Developer Edition

**Comment by pawan-poudel-github** [comment]
[Shadcn](https://ui.shadcn.com) has added drawer component recently. you can use that instead of this. it just work fine

**Comment by mrabbas1** [comment]
thanks

**Comment by gclark-eightfold** [comment]
> [Shadcn](https://ui.shadcn.com) has added drawer component recently. you can use that instead of this. it just work fine

Shadcn's drawer is just a styled version of this package. It will have the same issues.
The issue seems to be the:
```typescript
    if (isAllowedToDrag.current && isInput(event.target as HTMLElement)) {
      // If we were just dragging, prevent focusing on inputs etc. on release
      (event.target as HTMLInputElement).blur();
    }
```
code in `onRelease`. Seems to be triggering this condition when the focus is on the input. For now I've just patched it locally to remove that block and it fixed the issue.

**Comment by mrabbas1** [comment]
> > [Shadcn](https://ui.shadcn.com) has added drawer component recently. you can use that instead of this. it just work fine
> 
> Shadcn's drawer is just a styled version of this package. It will have the same issues. The issue seems to be the:
> 
> ```ts
>     if (isAllowedToDrag.current && isInput(event.target as HTMLElement)) {
>       // If we were just dragging, prevent focusing on inputs etc. on release
>       (event.target as HTMLInputElement).blur();
>     }
> ```
> 
> code in `onRelease`. Seems to be triggering this condition when the focus is on the input. For now I've just patched it locally to remove that block and it fixed the issue.

@emilkowalski  i think It will be much better if you update

**Comment by GrantClark1999** [comment]
Just as a follow-up, I've updated the original sandbox to the latest version [here](https://codesandbox.io/p/devbox/drawer-with-scale-forked-spcfh6?file=%2Fapp%2Fmy-drawer.tsx%3A7%2C39)

In my own testing, I've only ever gotten it to just be a minor inconvenience, and was reproducible by rapidly clicking in & out of the input box to the left & right of it. I can record a video if it helps, but I figure it might not be super clear _when_ I'm clicking unless I'm using something like screenstudio.

Also, I wasn't able to figure out specifically why, but I had a bug report from a friend where trying to focus on the input field in the drawer never worked and would **always** blur before being able to type, making the form completely unusable.

I _believe_ it had something to do with their use of Google's autocomplete which would pop up everytime they focused the field, but I couldn't 100% confirm.

Either way, patching the above-mentioned code **does** fix the issue both for me and my friend who has the unusable input fields.

**Comment by danielberndt** [comment]
Not all clicks are perfect. Depending on the device and the user, the pointer might move a few pixels between the pointer down and the pointer up events. The `onRelease` code referenced in the snippet above interprets this as a drag event.
Probably would be good to allow a few pixels of slack before triggering the `blur()` event.

**Comment by KingXP-Pythoner** [comment]
I can't believe I've lost over 3 days debugging and the vaul drawer was the issue. Really hope this gets fixed as it's practically impossible to use the components with forms

**Comment by KingXP-Pythoner** [comment]
> Just as a follow-up, I've updated the original sandbox to the latest version [here](https://codesandbox.io/p/devbox/drawer-with-scale-forked-spcfh6?file=%2Fapp%2Fmy-drawer.tsx%3A7%2C39)
> 
> In my own testing, I've only ever gotten it to just be a minor inconvenience, and was reproducible by rapidly clicking in & out of the input box to the left & right of it. I can record a video if it helps, but I figure it might not be super clear _when_ I'm clicking unless I'm using something like screenstudio.
> 
> Also, I wasn't able to figure out specifically why, but I had a bug report from a friend where trying to focus on the input field in the drawer never worked and would **always** blur before being able to type, making the form completely unusable.
> 
> I _believe_ it had something to do with their use of Google's autocomplete which would pop up everytime they focused the field, but I couldn't 100% confirm.
> 
> Either way, patching the above-mentioned code **does** fix the issue both for me and my friend who has the unusable input fields.

Can i ask how you're able to patch the package? In my node_modules, theres only a dist folder with uglified code instead of the original code.

**Comment by emilkowalski** [comment]
This has been fixed in #244.
